File: kb-engine/tests/fixtures/text-ingest/sample.py
"""Sample Python file for text-ingest tests.

Two top-level functions, one class with two methods. The chunker should
emit one chunk per top-level definition (the class is one chunk regardless
of how many methods it contains).
"""
from __future__ import annotations


def add(a: int, b: int) -> int:
    """Return the sum of two ints."""
    return a + b


def multiply(a: int, b: int) -> int:
    """Return the product of two ints."""
    return a * b


class Calculator:
    """A trivial stateful calculator used to exercise class chunking."""

    def __init__(self, start: int = 0) -> None:
        self.value = start

    def add(self, x: int) -> int:
        self.value += x
        return self.value

    def reset(self) -> None:
        self.value = 0

File: kb-engine/tests/fixtures/text-ingest/sample.ts
/**
 * Sample TypeScript file for text-ingest tests.
 *
 * Contains: a named function, an interface, and an exported arrow function.
 * The chunker should emit one chunk per top-level declaration.
 */

export function greet(name: string): string {
  return `hello, ${name}`;
}

export interface User {
  id: string;
  email: string;
  createdAt: Date;
}

export const formatUser = (u: User): string => {
  return `${u.email} (${u.id})`;
};
